### PR TITLE
Discuss the concept of superconvergence better.

### PR DESCRIPTION
**Pull request by bangerth**
Following up on a discussion on the mailing list with @cpraveen .

### CODE REVIEW

File: examples/step-7/step-7.cc [hunk 1]
```diff
@@ -868,15 +868,30 @@ namespace Step7
...
+    // it. It turns out that what we get depends quite sensitively on the
+    // "quadrature" points being used. There is also the issue of
+    // superconvergence: Finite element solutions are, on some meshes and for
+    // polynomial degrees $k\ge 2$, particularly accurate at the node points as
+    // well as at Gauss points, much more accurate in any case than at randomly
+    // chosen points. (See
+    // @cite Li2019 and the discussion and references in Section 1.2 for more
+    // information on this.) In other words, if we are interested in finding
+    // the largest difference $u(\mathbf x)-u_h(\mathbf x)$, then we ought to
+    // look at point $\mathbf x$ that are specifically not of this "special"
```
Comment (tjhei) [inline examples/step-7/step-7.cc:887]:
```suggestion
    // look at points $\mathbf x$ that are specifically not of this "special"
```

File: examples/step-7/step-7.cc [hunk 2]
```diff
@@ -868,15 +868,30 @@ namespace Step7
...
+    // kind of points -- in other words, we should specifically not use
+    // `QGauss(fe->degree+1)` to define where we evaluate. Rather, we use a
+    // special quadrature rule that is obtained by iterating the trapezoidal
+    // rule by the degree of the finite element times two plus one in each space
+    // direction. Note that the constructor of the QIterated class takes a
+    // one-dimensional quadrature rule and a number that tells it how often it
+    // shall use this rule in each space direction.
```
Comment (tjhei) [inline examples/step-7/step-7.cc:894]:
```suggestion
    // shall repeat this rule in each space direction.
```
maybe?

File: examples/step-7/step-7.cc [hunk 3]
```diff
@@ -868,15 +868,30 @@ namespace Step7
...
-    // takes a one-dimensional quadrature rule and a number that tells it how
-    // often it shall use this rule in each space direction.
+    // compute the true maximum of the error over *all* points in the domain,
+    // but only the maximum over a finite set of evaluation points that, for
+    // convenience, we will still call "quadrature points" and represent by
+    // an object of type Quadrature) even though we do not actually perform any
+    // integration.
+    //
+    // There is then the question of what points precisely we want to evaluate
+    // it. It turns out that what we get depends quite sensitively on the
```
Comment (tjhei) [inline examples/step-7/step-7.cc:878]:
```suggestion
    // at. It turns out that the result we get depends quite sensitively on the
```

File: examples/step-7/step-7.cc [hunk 4]
```diff
@@ -868,15 +868,30 @@ namespace Step7
...
+    // "quadrature" points being used. There is also the issue of
+    // superconvergence: Finite element solutions are, on some meshes and for
+    // polynomial degrees $k\ge 2$, particularly accurate at the node points as
+    // well as at Gauss points, much more accurate in any case than at randomly
```
Comment (tjhei) [inline examples/step-7/step-7.cc:882]:
```suggestion
    // well as at Gauss points, much more accurate than at randomly
```

File: examples/step-7/step-7.cc [hunk 5]
```diff
@@ -868,15 +868,30 @@ namespace Step7
...
+    // chosen points. (See
+    // @cite Li2019 and the discussion and references in Section 1.2 for more
+    // information on this.) In other words, if we are interested in finding
+    // the largest difference $u(\mathbf x)-u_h(\mathbf x)$, then we ought to
+    // look at point $\mathbf x$ that are specifically not of this "special"
+    // kind of points -- in other words, we should specifically not use
```
Comment (tjhei) [inline examples/step-7/step-7.cc:888]:
```suggestion
    // kind of points and we should specifically not use
```

File: examples/step-7/step-7.cc [hunk 6]
```diff
@@ -868,15 +868,30 @@ namespace Step7
...
-    // super-convergence effects at some points, we use a special quadrature
-    // rule that is obtained by iterating the trapezoidal rule by the degree of
-    // the finite element times two plus one in each space direction.
-    // Note that the constructor of the QIterated class
-    // takes a one-dimensional quadrature rule and a number that tells it how
-    // often it shall use this rule in each space direction.
+    // compute the true maximum of the error over *all* points in the domain,
+    // but only the maximum over a finite set of evaluation points that, for
+    // convenience, we will still call "quadrature points" and represent by
+    // an object of type Quadrature) even though we do not actually perform any
```
Comment (marcfehling) [inline examples/step-7/step-7.cc:874]:
```suggestion
    // an object of type Quadrature even though we do not actually perform any
```
I couldn't find the counterpart.

File: doc/doxygen/references.bib [hunk 7]
```diff
@@ -26,6 +26,23 @@ @article{Kel74
...
+@article{Li2019,
+  doi = {10.1007/s10915-019-01102-1},
+  url = {https://doi.org/10.1007/s10915-019-01102-1},
+  year = {2019},
+  month = dec,
+  publisher = {Springer Science and Business Media {LLC}},
+  volume = {82},
+  number = {1},
+  author = {Hao Li and Xiangxiong Zhang},
+  title = {Superconvergence of {\textdollar}{\textdollar}C{\^{}}0{\textdollar}{\textdollar}C0-{\textdollar}{\textdollar}Q{\^{}}k{\textdollar}{\textdollar}Qk Finite Element Method for Elliptic Equations with Approximated Coefficients},
```
Comment (marcfehling) [inline doc/doxygen/references.bib:42]:
```suggestion
  title = {Superconvergence of $C^0-Q^k$ Finite Element Method for Elliptic Equations with Approximated Coefficients},
```
This looks odd. My suggestion is based on 'Brenner 2005` later in that document.

**Comment by bangerth** [comment]
All good points -- thanks, guys! Fixed.

File: examples/step-7/step-7.cc [hunk 8]
```diff
@@ -868,15 +868,30 @@ namespace Step7
...
+    // but only the maximum over a finite set of evaluation points that, for
+    // convenience, we will still call "quadrature points" and represent by
+    // an object of type Quadrature even though we do not actually perform any
+    // integration.
+    //
+    // There is then the question of what points precisely we want to evaluate
+    // at. It turns out that the result we get depends quite sensitively on the
+    // "quadrature" points being used. There is also the issue of
+    // superconvergence: Finite element solutions are, on some meshes and for
+    // polynomial degrees $k\ge 2$, particularly accurate at the node points as
```
Comment (drwells) [inline examples/step-7/step-7.cc:881]: Its not worth changing, but you still get order 2k convergence for linear elements: its just that k + 1 = 2 * k in that case :)

File: examples/step-7/step-7.cc [hunk 9]
```diff
@@ -868,15 +868,30 @@ namespace Step7
...
+    // well as at Gauss points, much more accurate than at randomly
```
Comment (drwells) [inline examples/step-7/step-7.cc:882]:
```suggestion
    // well as at Gauss-Lobatto points, much more accurate than at randomly
```

**Comment by drwells** [comment]
@bangerth *ping*

**Comment by bangerth** [comment]
@drwells comment addressed.

**Review by drwells — APPROVED** [review]

**Review by masterleinad — APPROVED** [review]

**Pull request merged**